lib/ansible/modules/files/copy.py

```diff
@@ -45,14 +45,15 @@
   content:
     version_added: "1.1"
     description:
-      - When used instead of 'src', sets the contents of a file directly to the specified value.
+      - When used instead of I(src), sets the contents of a file directly to the specified value.
         This is for simple values, for anything complex or with formatting please switch to the template module.
     required: false
     default: null
   dest:
     description:
-      - Remote absolute path where the file should be copied to. If src is a directory,
-        this must be a directory too.
+      - Remote absolute path where the file should be copied to. If I(src) is a directory, this must be a directory too and
+        I(dest) directory is created when it doesn't exist.
+        If I(src) and I(dest) are files, then parent directory of I(dest) isn't created: the task fails if it doesn't already exist.
     required: true
```

Comment:
> I also notice the old "If I(src) is a directory, this must be a directory too" is incomplete. Ansible refuses to replace any existing <del>directories with a file</del> files with a directory, whether they are the top-level dest or a sub-directory. This is sensible behaviour and matches unix cp.

Huh.  Existing _directories_ always cause the file to be copied into the directory, whether they are the top-level dest or a sub-directory.  This is more clear-cut as a bug.  Especially, because if you create a directory inside it as well, which blocks the file, it fails with "attempted to take checksum of directory".  Filed as #25792